src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -185,11 +185,37 @@ protected internal static void ValidateMatchTimeout(TimeSpan matchTimeout)
             }
         }
 
-        protected Regex(SerializationInfo info, StreamingContext context) =>
-            throw new PlatformNotSupportedException();
+        protected Regex(SerializationInfo info, StreamingContext context)
+        {
+            SerializationInfoEnumerator infoEnum = info.GetEnumerator();
+            SRM.Regex srmregex = null;
+            while (infoEnum.MoveNext())
```

Comment:
In this context perhaps the code can assume that the first entry in the SerializationInfo is the SRM blob, or it's not there at all.

But I'm not sure about using the ISerialization mechanism anyway (don't feel strongly)

Reply:
Right, then it does not really matter what the name of the key is.
The code would then be if (infoEnum.MoveNext()) .. use the Current.Value (if it is a string) throw exception. I'll update that.

Reply:
Regarding Dan's questions below. 

CredScan would use the deserialization and store them separately, as it does currently, for around 150 regexes or so for a fast deployment from scratch (in some key scenarios) to avoid an upfront cost  -- when the regexes have not been cached or loaded yet -- so the runtime is not up and running running yet.

However, the above represents a rather unusual scenario, but covers what they need/or have needed so far. (The drawback is that the particular serialization format is exposed.)
It would make **total sense** to also cache the serialization for a given (regex,options) pair for SRM so that if a DFA regex has already been seen it would just be deserialized -- as part of the existing regex caching system (modified a bit for the srm case).

However, all that said, we need to reevaluate the improved startup cost. I eliminated several bottlenecks (that had to do with use of earlier BSTs, perhaps minterms was not the only culprit there). But I still expect the cost to be considerable compared to deserialization that should be just nanosecs even for a reasonably large regex.

Regarding cost calculations. I think there are two scenarios. 1) Seeing what the initial "one off" cost is for, say say setting up all the 150 or so CredScan regexes (to pick a concrete example), i.e., the initial serialization cost compared to just deserializing them. 2) But also caching the DFA regex object and reusing it when possible (this is TBD): essentially caching
(pattern,options) -> srmregex.serialize(), when DFA in options, an emasuring what impact this has overall when reusing same regexes.

I can evaluate (1) and report on that. (I have problems compiling in release mode though from the VS solution). I was also considering putting the credscan regexes into a unit test, BUT: **CredScan team does not want these regexes to be public**. 
To doublecheck before I do that: I assume the test framework code is not public!!!